Automattic/woocommerce-payments
# Dev - Upgrade to Node v20

Fixes #9116

#### Changes proposed in this Pull Request

- Upgrade Node version to v20
- Upgrade packages that require a specific version of node. Most notably: `@woocommerce/components` and `@wordpress/scripts`. Other packages related to testing and linting were also updated.
- Update test snapshots
- Replace deprecated functions in tests
- Perform various fixes in tests so they work properly with the new versions of the packages that were upgraded.
- E2E Tests are being fixed in this other PR https://github.com/Automattic/woocommerce-payments/pull/9132

#### Testing instructions

* Make sure npm scripts work as expected locally: `npm build`, `npm start`, `npm lint`, `npm test:js`
* Make sure all checks are passing in the PR: css, lint, js, php
* You can also test the build in a Jurassic Ninja site; however, since no client code was changed, we don't expect any changes in build size and functionality.

-------------------


**Post merge**
